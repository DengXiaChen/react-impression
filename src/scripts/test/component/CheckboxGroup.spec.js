import React from 'react';
import { shallow } from 'enzyme';
import { Checkbox, CheckboxGroup } from '../../components/impression';

const setup = () => {
    const props = {
        defaultValue: ['basketball', 'football'],
        onChange: jest.fn(),
    };

    const wrapper = shallow(
        <CheckboxGroup {...props}>
            <Checkbox>basketball</Checkbox>
            <Checkbox>football</Checkbox>
            <Checkbox>volleyball</Checkbox>
        </CheckboxGroup>
    );

    return {
        props,
        wrapper,
    };
};

describe('CheckboxGroup', () => {
    const { wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.checkbox-inline').exists()).toBe(true);
        expect(wrapper.find('Checkbox')).toHaveLength(3);
    });

    it('defaultValue', () => {
        expect(wrapper.find('Checkbox').at(0).props().checked).toBe(true);
        expect(wrapper.find('Checkbox').at(1).props().checked).toBe(true);
        expect(wrapper.find('Checkbox').at(2).props().checked).toBe(false);
    });

    it('onChange', () => {
        wrapper.find('Checkbox').at(2).simulate('click');
        // TODO FIX
        // expect(props.onChange).toHaveBeenCalled();
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
