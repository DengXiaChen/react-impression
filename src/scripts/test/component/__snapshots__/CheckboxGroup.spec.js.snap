// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="checkbox-inline"
    defaultValue={
        Array [
            "basketball",
            "football",
          ]
    }
    disabled={false}
>
    <Checkbox
        checked={true}
        disabled={false}
        onChange={[Function]}
        value="basketball"
    >
        basketball
    </Checkbox>
    <Checkbox
        checked={true}
        disabled={false}
        onChange={[Function]}
        value="football"
    >
        football
    </Checkbox>
    <Checkbox
        checked={false}
        disabled={false}
        onChange={[Function]}
        value="volleyball"
    >
        volleyball
    </Checkbox>
</div>,
  "nodes": Array [
    <div
      className="checkbox-inline"
      defaultValue={
            Array [
                  "basketball",
                  "football",
                ]
      }
      disabled={false}
>
      <Checkbox
            checked={true}
            disabled={false}
            onChange={[Function]}
            value="basketball"
      >
            basketball
      </Checkbox>
      <Checkbox
            checked={true}
            disabled={false}
            onChange={[Function]}
            value="football"
      >
            football
      </Checkbox>
      <Checkbox
            checked={false}
            disabled={false}
            onChange={[Function]}
            value="volleyball"
      >
            volleyball
      </Checkbox>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <CheckboxGroup
        defaultValue={
                Array [
                        "basketball",
                        "football",
                      ]
        }
        direction="row"
        disabled={false}
        onChange={[Function]}
>
        <Checkbox
                disabled={false}
        >
                basketball
        </Checkbox>
        <Checkbox
                disabled={false}
        >
                football
        </Checkbox>
        <Checkbox
                disabled={false}
        >
                volleyball
        </Checkbox>
</CheckboxGroup>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CheckboxGroup {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "isPuppet": false,
        "onChangeHandle": [Function],
        "props": Object {
          "children": Array [
            <Checkbox
              disabled={false}
>
              basketball
</Checkbox>,
            <Checkbox
              disabled={false}
>
              football
</Checkbox>,
            <Checkbox
              disabled={false}
>
              volleyball
</Checkbox>,
          ],
          "defaultValue": Array [
            "basketball",
            "football",
          ],
          "direction": "row",
          "disabled": false,
          "onChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "value": Array [
            "basketball",
            "football",
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="checkbox-inline"
          defaultValue={
                    Array [
                              "basketball",
                              "football",
                            ]
          }
          disabled={false}
>
          <Checkbox
                    checked={true}
                    disabled={false}
                    onChange={[Function]}
                    value="basketball"
          >
                    basketball
          </Checkbox>
          <Checkbox
                    checked={true}
                    disabled={false}
                    onChange={[Function]}
                    value="football"
          >
                    football
          </Checkbox>
          <Checkbox
                    checked={false}
                    disabled={false}
                    onChange={[Function]}
                    value="volleyball"
          >
                    volleyball
          </Checkbox>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="checkbox-inline"
          defaultValue={
                    Array [
                              "basketball",
                              "football",
                            ]
          }
          disabled={false}
>
          <Checkbox
                    checked={true}
                    disabled={false}
                    onChange={[Function]}
                    value="basketball"
          >
                    basketball
          </Checkbox>
          <Checkbox
                    checked={true}
                    disabled={false}
                    onChange={[Function]}
                    value="football"
          >
                    football
          </Checkbox>
          <Checkbox
                    checked={false}
                    disabled={false}
                    onChange={[Function]}
                    value="volleyball"
          >
                    volleyball
          </Checkbox>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CheckboxGroup
    defaultValue={
        Array [
            "basketball",
            "football",
          ]
    }
    direction="row"
    disabled={false}
    onChange={[Function]}
>
    <Checkbox
        disabled={false}
    >
        basketball
    </Checkbox>
    <Checkbox
        disabled={false}
    >
        football
    </Checkbox>
    <Checkbox
        disabled={false}
    >
        volleyball
    </Checkbox>
</CheckboxGroup>,
}
`;
